sdk/src/main/java/io/opentelemetry/sdk/metrics/Batchers.java

```diff
@@ -47,5 +86,69 @@ public void batch(LabelSet labelSet, Aggregator aggregator, boolean mappedAggreg
     }
   }
 
+  private static final class AllLabels implements Batcher {
+    private final Descriptor descriptor;
+    private final Resource resource;
+    private final InstrumentationLibraryInfo instrumentationLibraryInfo;
+    private final Clock clock;
+    private final AggregatorFactory aggregatorFactory;
+    private Map<Map<String, String>, Aggregator> aggregatorMap;
+    private long startEpochNanos;
+    private final boolean delta;
+
+    private AllLabels(
+        Descriptor descriptor,
+        Resource resource,
+        InstrumentationLibraryInfo instrumentationLibraryInfo,
+        AggregatorFactory aggregatorFactory,
+        Clock clock,
+        boolean delta) {
+      this.descriptor = descriptor;
+      this.resource = resource;
+      this.instrumentationLibraryInfo = instrumentationLibraryInfo;
+      this.clock = clock;
+      this.aggregatorFactory = aggregatorFactory;
+      this.delta = delta;
+      this.aggregatorMap = new HashMap<>();
+      startEpochNanos = clock.now();
+    }
+
+    @Override
+    public final Aggregator getAggregator() {
+      return aggregatorFactory.getAggregator();
+    }
+
+    @Override
+    public final void batch(LabelSet labelSet, Aggregator aggregator, boolean unmappedAggregator) {
+      Map<String, String> labels = ((LabelSetSdk) labelSet).getLabels();
+      Aggregator currentAggregator = aggregatorMap.get(labels);
+      if (currentAggregator == null) {
+        // This aggregator is not mapped, we can use this instance.
+        if (unmappedAggregator) {
+          aggregatorMap.put(labels, aggregator);
+          return;
+        }
+        currentAggregator = aggregatorFactory.getAggregator();
+        aggregatorMap.put(labels, currentAggregator);
+      }
+      aggregator.mergeToAndReset(currentAggregator);
+    }
+
+    @Override
+    public final List<MetricData> completeCollectionCycle() {
+      List<Point> points = new ArrayList<>(aggregatorMap.size());
+      long epochNanos = clock.now();
+      for (Map.Entry<Map<String, String>, Aggregator> entry : aggregatorMap.entrySet()) {
```

Comment:
I think there's a risk of a ConcurrentModificationException happening during this iteration, since the map can be mutated during this method call.

Reply:
It is guaranteed that only one thread calls into a Batcher collection at a time by the collectLock in the AbstractInstrumentWithBindings

Reply:
please add a comment in here to that effect. thanks!